doc/SONiC_QoS_Mgmt_HLD-1.0.md

```diff
@@ -0,0 +1,499 @@
+````
+  sonic(config)# qos-map dscp-to-tc-map <name>
+  sonic(conf-qos-map)# traffic-class <1..8> dscp  {<dscp-value>}
+
+  sonic(config)# no dscp-to-tc-map <name>
+````
+
+- Config Traffic Class to Queue map
+````
+  sonic(config)# qos-map tc-to-queue-map <name>
+  sonic(conf-qos-map)# queue 3 traffic-class 0-3
+````
+
+- Config interface QoS
+````
+  sonic(config)# interface <name>
+  sonic(conf-if-name)# dscp-to-tc-map <name>
+  sonic(conf-if-name)# tc-to-queue-map <name>
+  sonic(conf-if-name)# scheduler-policy <name>
+  sonic(conf-if-name)# queue <q#> wred-profile <wred-profile-name>
+  sonic(conf-if-name)# queue <q#> [ucast|mcast] threshold <val> 
+     
+  sonic(conf-if-name)# no dscp-to-tc-map <name>
+  sonic(conf-if-name)# no tc-to-queue-map <name>
+  sonic(conf-if-name)# no scheduler-policy <name>
+  sonic(conf-if-name)# no queue <q#> wred-profile <wred-profile-name>
+  sonic(conf-if-name)# no queue <q#> [ucast|mcast] threshold 
+
+````
+- Config Watermark polling interval
+````
+  sonic(config)# qos watermark polling-interval <val>
+````
+
+
+#### 3.2.2.2 Show Commands
+
+- Show WRED
+````
+  >show wred {<name>}
+  Sample Output:
+     > show wred wred_1
+    Profile Name | Green            | Yellow           | Red              |       |    
+                 |------------------|------------------|------------------|       |    
+                 | MIN MAX DROP-RATE| MIN MAX DROP-RATE| MIN MAX DROP-RATE|  ECN  |
+                 | KB  KB  %        | KB  KB  %        | KB  KB  %        |       |  
+    -------------|------------------|------------------|------------------|-------|
+      wred_1     | 100 1000 100     | 50 100 100       | 50 100 100       | All   |
+    -------------|------------------|------------------|------------------|-------|
+````
+
+
+- Show Scheduler Policy  
+````    
+  >show scheduler-policy <name>   
+  Sample output:
+    Scheduler <q0>:
+      Type: Priority
+      Bandwidth Percent: N/A
+      Priority Level: 1
+      Shape Bandwidth: min 100mbps, min-burst 16kBytes, max 200mbps, max-burst 16kBytes
+    Scheduler <q1>:
+      Type: WRR
+      Bandwidth Percent: 40
+      Shape Bandwidth: min 200mbps, min-burst 16kBytes, max 1000mbps, max-burst 16kBytes
+    ...
+````	
+
+
+- Show interface QoS configuration
+````
+  >show qos interface <name>  	
+  Sample Output:
+      Ethernet 1/1/10
+      DSCP-to-Traffic-Class map: dscp-to-tc-map_1
+      Traffic-Class-to-Queue map: tc-to-queue-map_1
+      Scheduler policy: scheduler-policy_1
+      Q1 WRED profile:  wred-profile_1
+      Q5 WRED profile:  wred-profile_2
+      Q1 threshold: 2500
+      Q5 threshold: 2500 
+````
+
+- Show DSCP to TC map
+````
+  >show qos-map dscp-to-tc-map <name>   
+  Sample Output:
+	DSCP Priority to Traffic-Class Map : dscp-trustmap1
+        Traffic-Class | DSCP Priority
+        --------------|---------------
+         0            |	8-15
+         2            |	16-23
+         1            |	0-7

```

Comment:
Since the mapping is DSCP to traffic class, we can have DSCP on left and Traffic Class on right.
Also please add example of mapping where DSCP are not in continuous range e.g. DSCP 4,10-20,25 mapped to traffic-class 4